src/SimpleGraphs/simpledigraph.jl

```diff
@@ -246,26 +246,47 @@ is_directed(g::SimpleDiGraph) = true
 is_directed(::Type{SimpleDiGraph}) = true
 is_directed(::Type{SimpleDiGraph{T}}) where T = true
 
+
+function has_edge(g::SimpleDiGraph{T}, e::SimpleDiGraphEdge{T}) where T
+    s, d = T.(Tuple(e))
+    (s in vertices(g) && d in vertices(g)) || return false  # edge out of bounds
+    @inbounds list = g.fadjlist[s]
+    index = searchsortedfirst(list, d)
+    @inbounds return (index <= length(list) && list[index] == d)
+end
```

Comment:
The advantage of the existing method is that we check against the shorter of the forward or the backward adjacency list. I would recommend doing the same thing here:
```
list, v = length(g.fadjlist[s]) < length(g.badjlist[d]) ? g.fadjlist[s], d : g.badjlist[d], s
index = searchsortedfirst(list, v)
```

Reply:
Yeah, I was not sure about. I will add the case distinction back although it will probably only make a difference if there is large difference in the degrees of the graph.

Reply:
> it will probably only make a difference if there is large difference in the degrees of the graph.

Yep, that's precisely why we have it. When dealing with large directed social graphs, it's a huge time saver, and the extra comparison doesn't cost much on smaller graphs.